### DISCUSSION TITLE
difference between dart port and python pkg

### DISCUSSION THREAD

**Discussion by 2bndy5** [cpp-linter/cpp-linter#60] [Ideas]
There are some differences in this port compared to the python source. To start this discussion, I'll note the major difference.

# CLI args

Some of the CLI options are actually implemented as switches (AKA flags) where the dart arguments library automatically adds a inverse for a specified flag:

This enables the flag
```
cpp-linter --file-annotations
```
While this can disable the flag
```
cpp-linter --no-file-annotations
```

I have an idea about how to work with this in the action.yml for cpp-linter-action repo, but we may not actually use this experimental source at all.

# Lines changes only feature

This dart port only focuses on the lines with actual additions. The python source originally focused on all lines in the diff, but because of some confusion, this led to a change in the default behavior.

I am storing the data needed for both approaches in dart, but I found it easier to only focus on lines with additions. Thus, there is some dead code here.

---
We can talk more about others as we go, but I just wanted a place where we _could_ discuss. I expect this discussion will lead to improvements in the python source.

**Comment by 2bndy5** [comment]
Overall I'm not very impressed with dart's unit testing convention. It may be so new to me that I just need time to adapt.

**Reply by shenxianpeng** [reply]
Would you continue to work on cpp_linter_dart or it has already been replaced by cpp-linter-rs and it can be archived?

**Reply by 2bndy5** [reply]
It can be archived. With the new node.js bindings from cpp-linter-rs project, the cpp-linter-dart project is completely obsolete.

**Reply by shenxianpeng** [reply]
Done.
